### PR TITLE
CTCH-7069: patch Lettuce scan cursor handling around for unsigned Longs

### PR DESCRIPTION
**Pull request by mjohnston-il**
See https://github.com/spring-projects/spring-data-redis/commit/6e0f30b6076ef453f202d9cdcd0ba8c44365fc9a#diff-0cc7920f9925f6cdfee66ecb21f3ade6978fd28d3124298a0b10e0e979170947 for the source of these changes.

I didn't bother to patch the Jedis files because we don't use Jedis

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,11 +14,11 @@
 	<parent>
 		<groupId>org.springframework.data.build</groupId>
 		<artifactId>spring-data-parent</artifactId>
-		<version>2.7.19-SNAPSHOT</version>
+		<version>2.7.18</version>
```
Comment (matthewvasallo) [inline pom.xml:17]: Why was this needed?

Comment (mjohnston-il) [reply pom.xml:17]: I'm not sure that is, but it felt weird to have our builds rely on a Spring snapshot version

**Review by matthewvasallo — APPROVED** [review]
🚢
